components/sup/src/manager.rs

```diff
@@ -1256,13 +1256,24 @@ impl Manager {
             self.spec_dir
                 .specs()
                 .into_iter()
-                .filter(|spec| !existing_idents.contains(&spec.ident))
-                .flat_map(|spec| {
-                    Service::new(self.sys.clone(),
-                                 spec,
-                                 self.fs_cfg.clone(),
-                                 self.organization.as_ref().map(String::as_str),
-                                 self.state.gateway_state.clone()).into_iter()
+                .filter_map(|spec| {
+                    let ident = spec.ident.clone();
+                    if existing_idents.contains(&ident) {
+                        None
+                    } else {
+                        let service = Service::new(self.sys.clone(),
+                                                   spec,
+                                                   self.fs_cfg.clone(),
+                                                   self.organization.as_ref().map(String::as_str),
+                                                   self.state.gateway_state.clone());
+                        match service {
+                            Ok(service) => Some(service),
+                            Err(e) => {
+                                warn!("Failed to create service '{}' from spec: {:?}", ident, e);
+                                None
+                            }
+                        }
```

Comment:
If `service` is renamed (say to `result`, I think it makes the more concise version of the match block more readable:
```rust
                        if let Err(ref e) = result {
                            warn!("Failed to create service '{}' from spec: {:?}", ident, e);
                        }
                        result.ok()
```
I dislike the redundancy of `Ok(_) => Some(_)`, but maybe `Result::ok` is less clear. Use whichever you think is best.